### PR TITLE
Add UTM parameters to links in subscription renewal notices

### PR DESCRIPTION
**Pull request by thilinah**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds UTM parameters to subscription renewal links in notices shown on WC settings, in-app extensions and plugins page.

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Build WooCommerce on your local environment, e.g.: `cd plugins/woocommerce` then `npm run build:zip`.
- Update the WooCommerce plugin in your local store with WC build in previous step.
- Connected your store with woocommerce.test.
- Ensure Woo Update Manager is installed.

### Test Single Expired Subscription

- Ensure you have only one expired product key in woocommerce.test.
- Visit in-app my subscriptions tab (`/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`)
- There should be a message like bellow and the renew button on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_in_apps_screen_renew
```

Screenshot 2024-07-17 at 19 56 29

- Visit  WC Settings
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_settings_screen_renew
```
Screenshot 2024-07-17 at 20 15 14

- Visit plugin management page 
- Make sure the plugin version is below the current version by changing it manually on the plugin main file.
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_plugin_screen_renew
```
Screenshot 2024-07-17 at 20 21 09

### Test Multiple Expired Subscriptions of different products

- Ensure you have 2 or more  expired product keys in woocommerce.test.
- Visit in-app my subscriptions tab (`/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`)
- There should be a message like bellow and the renew button on the notice should have UTM parameters: 
```
utm_source=pu 
utm_campaign=pu_in_apps_screen_renew
```
Screenshot 2024-07-17 at 20 30 46

- Visit WC settings page
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_settings_screen_renew
```
Screenshot 2024-07-17 at 20 32 26

### Test Multiple Expired Subscriptions of the same product

- Ensure you have 2 or more  expired product keys of same product (and no other expired keys) in woocommerce.test.
- Visit in-app my subscriptions tab (`/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`)
- There should be a message like bellow and the renew button on the notice should have UTM parameters: 
```
utm_source=pu 
utm_campaign=pu_in_apps_screen_renew
```
Screenshot 2024-07-17 at 20 45 55

- Visit WC settings page
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_settings_screen_renew
```
Screenshot 2024-07-17 at 20 49 23

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Add UTM parameters to subscription renewal notice links.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php [hunk 1]
```diff
@@ -272,21 +274,37 @@ public static function display_notice_for_expired_and_expiring_subscriptions( $p
 		// Prepare the expiry notice based on subscription status.
 		$expiry_notice = '';
 		if ( ! empty( $subscription['expired'] ) && ! $subscription['lifetime'] ) {
+
+			$renew_link = add_query_arg(
+				array(
+					'utm_source'   => 'pu',
+					'utm_campaign' => 'pu_plug_screen_renew',
```
Comment (hesahesa) [inline plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php]: maybe this should be `pu_plugin_screen_renew`?

Comment (thilinah) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php]: Updated with https://github.com/woocommerce/woocommerce/pull/49645/commits/d048db42d574aade5a61664dd4093f8e835cc734

**Comment by gedex** [comment]
**Single expired**

* [x] Test single expired subscription in in-apps pages
* [x] Test single expired subscription in wc setting pages
* [x] Test single expired subscription in plugin page

**Multi expired (different products)**

* [x] Test multi expired subscriptions (different products) in in-apps pages
* [x] Test multi expired subscriptions (different products) in wc setting pages
* [x] Test multi expired subscriptions (different products) in plugin page

**Multi expired (same product)**

* [x] Test multi expired subscriptions (same product) in in-apps pages
* [x] Test multi expired subscriptions (same product) in wc setting pages
* [x] Test multi expired subscriptions (same product) in plugin page

**Review by gedex — APPROVED** [review]
LGTM. Tested and works as described!

**Pull request merged**